src/fu-keyring-utils.c

```diff
@@ -100,28 +100,17 @@ fu_keyring_get_release_trust_flags (AsRelease *release,
 		{ FWUPD_KEYRING_KIND_NONE,	NULL }
 	};
 
-	/* no filename? */
-	csum_tmp = as_release_get_checksum_by_target (release, AS_CHECKSUM_TARGET_CONTENT);
-	if (csum_tmp == NULL) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INVALID_FILE,
-				     "no content checksum for release");
-		return FALSE;
-	}
-	fn = as_checksum_get_filename (csum_tmp);
-	if (fn == NULL) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INVALID_FILE,
-				     "no filename");
-		return FALSE;
-	}
+	/* custom filename specified */
+	fn = xb_node_query_attr (release, "checksum[@target='content']", "filename", NULL);
+	if (fn == NULL)
```

Comment:
This seems like a new behavior.  Or was this what appstream-glib was internally doing previously?

Reply:
Yes, somewhat. The daemon was setting the "filename.bin" filename on the AsChecksum object which was saved in the AsRelease, which was saved in the AsApp. It was "fixing up" missing information at runtime. The XbNodes of libxmlb are immutable, and so we can't fix up when we load the component, we have to fix it up when we actually need the filename.

The alternate is we do a XbBuilderNode fixup when parsing the original XML, but maybe that's a bit magic, and certainly more complicated than the two lines here.

Reply:
Yes, makes sense now thanks.